pinot-hadoop-filesystem/src/main/java/com/linkedin/pinot/filesystem/HadoopPinotFS.java

```diff
@@ -97,39 +109,93 @@ public boolean move(URI srcUri, URI dstUri, boolean overwrite) throws IOExceptio
    */
   @Override
   public boolean copy(URI srcUri, URI dstUri) throws IOException {
-    Path source = new Path(srcUri);
-    Path target = new Path(dstUri);
-    RemoteIterator<LocatedFileStatus> sourceFiles = _hadoopFS.listFiles(source, true);
-    if (sourceFiles != null) {
-      while (sourceFiles.hasNext()) {
-        boolean succeeded = FileUtil.copy(_hadoopFS, sourceFiles.next().getPath(), _hadoopFS, target, true, _hadoopConf);
-        if (!succeeded) {
-          return false;
+    if (!exists(srcUri)) {
+      LOGGER.warn("Source {} does not exist", srcUri);
+      return false;
+    }
+    if (srcUri.equals(dstUri)) {
```

Comment:
What if srcUri is a directory and has files that actually need to copied over?

Reply:
Do you mean there's an edge case that Line 116 can't handle? 
If src and dst are the same, regardless of src's type, nothing should be done.